### ISSUE TITLE
UPDATE ONLY and INSERT INTO ONLY for inherited tables in PostgreSQL

### ISSUE THREAD

**Issue by Kelthogar** [dbeaver/dbeaver#9519]
Hello,

In our project we have some inherited tables and we are currently testing DBeaver, because we look for a great database management tool.
Until now we are VERY happy with the features of DBeaver and of its intuitive usage, BUT we have discovered one problem with updating or inserting in our inherited tables in PostgreSQL.

For example, when we try to update a value, than our trigger of the history table throws an error, because he doesn't allow updates or inserts in the history table itself.
Thats logical, because DBeaver trys to "UPDATE table SET ...", but what DBeaver should do is an "UPDATE ONLY table SET ..." statement. Same thing for "INSERT INTO" instead of "INSERT INTO ONLY".

Is there a possibility that DBeaver can put the additional "ONLY" argument in his statements, when the user works with an inherited table?

With best regards,
Kelthogar

**Comment by kseniiaguzeeva** [comment]
Thank you for the suggestion

**Comment by serge-rider** [comment]
I don' think that we should enable this behavior by default.
As usually people assume that all inherited/children tables will be updated.

We may add some extra feature. E.g. ask user how exactly he/she wants to update table. Or add some checkbox-like trigger to switch the behavior.
WDYT?

**Comment by Kelthogar** [comment]
@serge-rider:
Yes, we think also that a checkbox-like trigger would be the smartest way to implement this feature. :)
